src/validations/rules/Writing_FedRAMP_Validations.md

```diff
@@ -0,0 +1,60 @@
+- [OSCAL](https://pages.nist.gov/OSCAL/) is used to define the several FedRAMP document types in a structured data form. NIST provides XML Schema documents for the various OSCAL document types.
+- [Schematron](https://schematron.com/) is used to declare assertions for documents to be validated
+- [XPath](https://www.w3.org/TR/xpath-31/) is used in Schematron to define rule contexts and assertion tests
+- [XSpec](XSpec) is used to creaste unit tests for Schematron assertions
+- [XSLT](https://www.w3.org/TR/xslt-30/) is used to manipulate XML documents, and is used to transpile Schematron documents into XSLT documents
+
+# FedRAMP OSCAL References
+
+FedRAMP has adopted and extended NIST OSCAL document types. These adaptations are documented in the GSA [fedramp_automation repository](https://github.com/GSA/fedramp-automation/tree/master/documents).
+
+The pertinent normative documents are
+
+-[Guide to OSCAL-based FedRAMP Content](./Guide_to_OSCAL-based_FedRAMP_Content.pdf)
+
+-[Guide to OSCAL-based FedRAMP System Security Plans (SSP)](./Guide_to_OSCAL-based_FedRAMP_System_Security_Plans_(SSP).pdf)
+
+-[Guide to OSCAL-based FedRAMP Security Assessment Plans  (SAP)](./Guide_to_OSCAL-based_FedRAMP_Security_Assessment_Plans_(SAP).pdf)
+
+-[Guide to OSCAL-based FedRAMP Security Assessment Reports (SAR)](./Guide_to_OSCAL-based_FedRAMP_Security_Assessment_Reports_(SAR).pdf)
+
+-[Guide to OSCAL-based FedRAMP Plan of Action and Milestones (POA&M)](./Guide_to_OSCAL-based_FedRAMP_Plan_of_Action_and_Milestones_(POAM).pdf)
+
+Each describes how FedRAMP expects each OSCAL document type is to be presented in an authorization package.
+
+# FedRAMP validation constraints
+
+FedRAMP validation constraints extend the NIST OSCAL XML Schema constraints. The XML Schema constraints are essentially syntactic; Schematron is used to augment the former with semantic constraints which cannot be expressed using XML Schema.
+
+Schematron constraints have thus far been for system security plan documents. They are found in a Schematron document `ssp.sch`.
+
+A companion document `fedramp_values.xml` defines sets of allowed values for various FedRAMP OSCAL XML  elements and their attributes.
+
+An additional Schmeatron document `sch.sch` defines constraints for the FedRAMP Schematron constraints themselves, guiding a FedRAMP constraint author to augment the fundamental Schematron with FedRAMP-specific constructs. These augmentations provide
+
+- References to associated unit tests
+- References to documentation related to the constraint
+- References to associated diagnostic messages
+- Forward and reverse linkages between Schematron rules (`<rule>` elements), assertions (`<assert>` elements, and `<diagnostic>` elements.
+
+# Examples of Schematron constraints
+
+## Example 1
+
+Consider this [code block](https://github.com/18F/fedramp-automation/blob/develop/src/validations/rules/ssp.sch#L2176-L2213).
```

Comment:
It may be preferable to include the code block in the markdown, rather than link to it - because the line numbers will quickly get out of date as `ssp.sch` is updated.

Reply:
Agreed.